tutor/src/screens/scores-report/late-work.jsx

```diff
@@ -9,43 +9,35 @@ import omit from 'lodash/omit';
 import { AsyncButton } from 'shared';
 import TaskResult from '../../models/course/scores/task-result';
 
-import TH from '../../helpers/task';
-
 class LateWorkMessages {
 
   displayName = 'LateWork';
 
   constructor(task) {
     this.task = task;
-    this.isAccepted = this.task.is_late_work_accepted;
-    this.status = this.isAccepted ?
-      TH.hasAdditionalLateWork(this.task) ? 'additional' : 'accepted'
-      :
-      'pending';
   }
 
-  score() {
-    if (this.status === 'accepted') {
-      return (
-        TH.getHumanUnacceptedScore(this.task)
-      );
-    } else {
-      return (
-        TH.getHumanScoreWithLateWork(this.task)
-      );
+  @computed get status() {
+    if (this.task.is_late_work_accepted) {
+      return this.task.hasAdditionalLateWork ? 'additional' : 'accepted';
     }
+    return 'pending';
+  }
+
+  @computed get isAccepted() {
+    return Boolean(
+      this.task.is_late_work_accepted && !this.task.hasAdditionalLateWork
+    );
+  }
+
+  score() {
+    return this.isAccepted ?
+      this.task.humanUnacceptedScore : this.task.humanScoreWithLateWork;
```

Comment:
I think this is still not right. It will fail if the instructor only accepted part of the late work. Should use the same formula as adjustScore()

Reply:
Hm, what these are doing is either showing what the score would be if the teacher accepts late work, or if they rejects the previously accepted work.  It's on the late work component and is used for the little popover that shows what it would be if the button's clicked.

Reply:
OK that makes sense then.